File: ios/Classes/listener/OnMsgSendCallback.swift

/**
 * 消息发送监听
 */
protocol OnMsgSendCallback: OnBase where T == Message {
    /**
     * 上传进度
     */
    func onProgress(progress: Int64)

    // Optionally, you can provide default implementations for the onSuccess and onError methods
    func onSuccess(data: Message) 

    func onError(code: Int, error: String) 
}

File: android/src/main/kotlin/io/openim/flutter_openim_sdk_ffi/listener/OnMsgSendCallback.kt
package io.openim.flutter_openim_sdk_ffi.listener

import io.openim.flutter_openim_sdk_ffi.models.Message

/**
 * 消息发送监听
 */
interface OnMsgSendCallback : OnBase<Message> {
    /**
     * 发送失败
     */
    override fun onError(code: Int, error: String)

    /**
     * 上传进度
     */
    fun onProgress(progress: Long)

    /**
     * 发送成功
     */
    override fun onSuccess(data: Message)
}